bugz.py

```diff
@@ -0,0 +1,202 @@
+import pandas as pd
+
+from lib.bugzilla_conn import BugzillaAPIClient
+
+from database import (
+    Database,
+    ReportBugzillaQEVerifyCount,
+    ReportBugzillaQENeeded
+)
+
+PRODUCTS = ["Fenix", "Focus", "GeckoView"]
+FIELDS = ["id", "summary", "flags", "severity",
+          "priority", "status", "resolution"]
+
+
+class Bugz:
+
+    def __init__(self) -> None:
+        self.conn = BugzillaAPIClient()
+
+    def get_bugs(self, bug_ids: list) -> list:
+        bugs = self.conn.bz_client.getbugs(bug_ids)
+        return bugs
+
+    def build_query(self, query: dict) -> dict:
+        formatted_query = self.conn.bz_client.build_query(query)
+        return formatted_query
+
+    def query(self, query: dict) -> list:
+        bugs = self.conn.bz_client.query(query)
+        return bugs
+
+    def get_query_from_url(self, url: str) -> dict:
+        query = self.conn.bz_client.url_to_query(url)
+        return query
+
+
+class BugzillaHelper:
+    def __init__(self) -> None:
+        self.bugzilla = Bugz()
+
+    def get_bugs(self, bugs: list) -> list:
+        """Get a list of bugs from Bugzilla."""
+        return self.bugzilla.get_bugs(bugs)
+
+    def build_query(self, query: dict) -> dict:
+        """Build a query for Bugzilla."""
+        return self.bugzilla.build_query(query)
+
+    def query(self, query: dict) -> list:
+        """Query Bugzilla."""
+        return self.bugzilla.query(query)
+
+    def get_query_from_url(self, url: str) -> dict:
+        """Get a query from a Bugzilla URL."""
+        return self.bugzilla.get_query_from_url(url)
+
+
+class BugzillaClient(Bugz):
+    def __init__(self):
+        super().__init__()
+        self.db = DatabaseBugzilla()
+        self.BugzillaHelperClient = BugzillaHelper()
+
+    def contains_criteria(self, entry, criteria):
+        return all(entry.get(key) == value for key, value in criteria.items())
+
+    def bugzilla_query(self):
+        allBugs = []
+        for product in PRODUCTS:
+            query = dict(product=product, include_fields=FIELDS)
+            bugs = self.BugzillaHelperClient.query(query)
+
+            for bug in bugs:
+                bug_ = [bug.id, bug.summary, bug.flags,
+                        bug.severity, bug.priority, bug.status, bug.resolution]
+                allBugs.append(bug_)
+
+        return allBugs
+
+    def bugzilla_query_qe_verify(self):
+        qe_bugs = []
+        search_criteria = {'name': 'qe-verify', 'status': '+'}
+
+        payload = self.bugzilla_query()
+
+        for bug in payload:
+            result = any(self.contains_criteria(entry, search_criteria) for entry in bug[2]) # noqa
```

Comment:
What is `bug[2]`?

Reply:
If we want to track the flags, we need to access the second element in the list given. This is how a bug looks like: 

[1909150, '[Menu Redesign] Implement accessibility reading order in the custom tab menu', [{'id': 2244803, 'setter': 'giorga@mozilla.com', 'type_id': 864, 'creation_date': <DateTime '20240917T09:39:02' at 0x147cb6cf0>, 'name': 'qe-verify', 'modification_date': <DateTime '20240917T09:39:02' at 0x147cb6d50>, 'status': '+'}], 'N/A', 'P2', 'RESOLVED', 'FIXED']

I am looking for the bugs that have that element and if there is the qe-verify and + values in it to filter the desired bugs